paasta_tools/spark_tools.py

```diff
@@ -253,10 +144,72 @@ def setup_volume_mounts(volumes: List[DockerVolume]) -> Dict[str, str]:
     return conf
 
 
+def create_spark_config_str(spark_config_dict, is_mrjob):
+    conf_option = "--jobconf" if is_mrjob else "--conf"
+    spark_config_entries = list()
+
+    if is_mrjob:
+        spark_master = spark_config_dict["spark.master"]
+        spark_config_entries.append(f"--spark-master={spark_master}")
+
+    for opt, val in spark_config_dict.items():
+        # mrjob use separate options to configure master
+        if is_mrjob and opt == "spark.master":
+            continue
+        # Process Spark configs with multiple space separated values to be in single quotes
+        if isinstance(val, str) and " " in val:
+            val = f"'{val}'"
+        spark_config_entries.append(f"{conf_option} {opt}={val}")
+    return " ".join(spark_config_entries)
+
+
 def inject_spark_conf_str(original_docker_cmd: str, spark_conf_str: str) -> str:
     for base_cmd in ("pyspark", "spark-shell", "spark-submit"):
         if base_cmd in original_docker_cmd:
             return original_docker_cmd.replace(
                 base_cmd, base_cmd + " " + spark_conf_str, 1
             )
     return original_docker_cmd
+
+
+def auto_add_timeout_for_spark_job(cmd, timeout_job_runtime):
+    # Timeout only to be added for spark-submit commands
+    # TODO: Add timeout for jobs using mrjob with spark-runner
+    if "spark-submit" not in cmd:
+        return cmd
+    try:
+        timeout_present = re.match(
+            r"^.*timeout[\s]+[\d]+[\.]?[\d]*[m|h][\s]+spark-submit .*$", cmd
+        )
+        if not timeout_present:
+            split_cmd = cmd.split("spark-submit")
+            cmd = f"{split_cmd[0]}timeout {timeout_job_runtime} spark-submit{split_cmd[1]}"
+            print(
+                PaastaColors.blue(
+                    f"NOTE: Job will exit in given time {timeout_job_runtime}. "
+                    f"Adjust timeout value using --timeout-job-timeout. "
+                    f"New Updated Command with timeout: {cmd}"
+                ),
+            )
+    except Exception as e:
+        err_msg = (
+            f"'timeout' could not be added to command: '{cmd}' due to error '{e}'. "
+            "Please report to #spark."
+        )
+        log.warn(err_msg)
+        print(PaastaColors.red(err_msg))
+    return cmd
+
+
+# TODO: Move to service config lib?
+def get_spark_ports(system_paasta_config: SystemPaastaConfig) -> Dict[str, int]:
+    return {
+        "spark.driver.port": system_paasta_config.get_spark_driver_port(),
+        "spark.blockManager.port": system_paasta_config.get_spark_blockmanager_port(),
+        "spark.driver.blockManager.port": system_paasta_config.get_spark_blockmanager_port(),
```

Comment:
These two are set to the same port `get_spark_blockmanager_port()`, which caused [Tron run failed](https://fluffy.yelpcorp.com/i/6bc1Td3kqSWjMgkHMc9Txdlw2w0Mkq6h.html).

We can remove all of these 3 port assignments, since the required preset ports are already set in service_configuration_lib:`get_spark_conf()`.

Reply:
Removed this